### PR TITLE
Add dfa.py

### PR DESCRIPTION
**Pull request by NthreeAhn**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [ x] **If creating a new file :**
  - [x ] added links to it in the README files ?
  - [x] included tests with it ?
  - [ ] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/automata/DFA.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        else:
+            cur = transitions[cur][string[i]]
+
+    for i in range (num_final):
+        if cur == final[i]:
+            return True
+        else:
+            return False
+        
+if __name__ == "__main__":
```
Comment (keon) [inline algorithms/automata/DFA.py]: please add the tests as unittests

**Comment by NthreeAhn** [comment]
Thank you for notice! Please check updated one.

File: tests/test_automata.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        transitions = {
+            'a': {'1': 'a', '0': 'b'},
+            'b': {'1': 'b', '0': 'a'}
+        }
+
+        final=['a']
+        start = 'a'
+    
+        self.assertEqual(False, DFA(transitions, start, final, "000111100"))
+        self.assertEqual(True, DFA(transitions, start, final, "111000011"))
```
Comment (keon) [inline tests/test_automata.py]: please add a bit more cases

Comment (NthreeAhn) [reply tests/test_automata.py]: I added two more cases. Thank you for your suggestion!!

**Pull request merged**